aea/protocols/dialogue/base.py

```diff
@@ -1026,6 +1051,46 @@ def __init__(self, dialogues: "Dialogues") -> None:
             {}
         )  # type: Dict[DialogueLabel, DialogueLabel]
         self._dialogues = dialogues
+        self._terminal_state_dialogues_labels: Set[DialogueLabel] = set()
+
+    @property
+    def dialogues_in_terminal_state(self) -> List["Dialogue"]:
+        """Get all dialogues in terminal state."""
+        return list(
+            filter(
+                None,
+                [
+                    self._dialogues_by_dialogue_label.get(i)
+                    for i in self._terminal_state_dialogues_labels
+                ],
+            )
+        )
+
+    @property
+    def dialogues_in_active_state(self) -> List["Dialogue"]:
+        """Get all dialogues in active state."""
+        active_dialogues = (
+            set(self._dialogues_by_dialogue_label.keys())
+            - self._terminal_state_dialogues_labels
+        )
+        return list(
+            filter(
+                None,
+                [self._dialogues_by_dialogue_label.get(i) for i in active_dialogues],
+            )
+        )
+
+    @property
+    def is_terminal_dialogues_kept(self) -> bool:
+        """Return True if dialogues should stay after terminal state."""
+        return self._dialogues.is_keep_dialogues_in_terminal_state
+
+    def dialogue_terminal_state_callback(self, dialogue: "Dialogue") -> None:
+        """Method to be called on dialogue terminal state reached."""
+        if self.is_terminal_dialogues_kept:
+            self._terminal_state_dialogues_labels.add(dialogue.dialogue_label)
+        else:
+            self.remove(dialogue.dialogue_label)
```

Comment:
remove currently does not remove dialogue label from _incomplete_to_complete_dialogue_labels, please fix

Reply:
done